### PR TITLE
Improve performance of evaluation in transfinite interpolation manifold

### PR DESCRIPTION
**Pull request by kronbichler**
This change inlines two functions from `GeometryInfo` for the default orientation case that would otherwise take up to 50% of the `compute_transfinite_interpolation` that is the core function of the `TransfiniteInterpolationManifold`.

Found while looking at #5368.

### CODE REVIEW

**Comment by drwells** [comment]
This is something that I noticed as well but never got around to fixing.

@kronbichler Would you mind sending me whatever benchmark you used to do this?

**Comment by kronbichler** [comment]
I did use your program from here
https://github.com/drwells/bad-laplace
but with the dimension changed to 3, a 3D ball mesh, used the `TransfiniteInterpolationManifold` on all cells, and put timers around the `integrate_difference()` call. The 50% number was obtained with valgrind's callgrind, though. In my tests I see something like 25 precent for larger mesh sizes. I send you an email with the code soon.

**Comment by drwells** [comment]
Thanks. At some point I would like to move more of the `GeometryInfo` lookup tables and functions into the header, so a benchmark like this is useful.

File: source/grid/manifold_lib.cc [hunk 1]
```diff
@@ -777,6 +777,33 @@ namespace
     return new_point;
   }
 
+  // this is replicated from GeometryInfo::face_to_cell_vertices since we need
+  // it very often in compute_transfinite_interpolation and the function is
+  // performance critical
+  unsigned int
+  face_to_cell_vertices_3d[6][4] =
+  {
+    {0, 2, 4, 6},
```
Comment (bangerth) [inline source/grid/manifold_lib.cc]: could you make this `const` or `static const`?

The code duplication is unfortunate. Can't we instead make the `GeometryInfo::face_to_cell_vertices` function `inline` and/or `constexpr`, to the same effect?

Comment (drwells) [reply source/grid/manifold_lib.cc]: Good point; that array really should be declared as `static constexpr`.

What we really need to do is add a faster path for `face_to_cell_vertices` that does not call `child_cell_on_face` (i.e., it skips all of the anisotropic logic).

Comment (kronbichler) [reply source/grid/manifold_lib.cc]: I agree, but the problem is that it really is hidden inside a whole layer of lookup tables. I don't want to pull them all to the header unconditionally, in particular not the one for `face_to_cell_vertices` which is a huge function (valgrind reported that we spent more than 120 instructions to get a single integer that I can do with 1 or 2 instructions with the lookup table). Along the suggestion by @drwells, my preference would be inline more. But I'd prefer to have the default case (standard orientation) where all we get is a single number from a single lookup table in the header and everything else in the `.cc` file. I do not have a simple strategy to implement that (that's why this PR looked the way it did), so I'd appreciate if someone wants to take the lead. As to make that `const`: I will do that in a follow-up PR with documentation updates once #5368 is merged.

Comment (bangerth) [reply source/grid/manifold_lib.cc]: I'm not terribly opposed to putting 200 lines of code into a header file if it has a demonstrable impact -- there are, after all, roughly 300,000 lines of code in the `include/` directory :-)

**Pull request merged**